docs/proposal/kmesh_support_localityLB.md

```diff
@@ -0,0 +1,183 @@
+
+### Goals
+
+The purpose of this proposal is to add locality-based topology-aware load balancing capabilities to kmesh workload mode, corresponding to locality load balancing in Ztunnel.
+
+### Proposal
+
+Locality load balancing mode: locality failover, locality strict.
+
+What is locality failover: When a flow accesses the service, the control plane will perform a tiered match between the locality information of the originating pod of the flow and the localities of all healthy backends behind the service. Pods with a higher degree of match indicate that they are geographically closer, and traffic will be preferentially routed to pods with a higher matching pair.
+
+What is locality strict: It is a mode within the locality failover mechanism. In locality failover, backends with no match at all are considered the lowest priority; in strict mode, backends that have absolutely no match are eliminated and traffic is not routed to these unmatched backends at all.
+
+### Design Details
+
+#### configure
+ref: https://pkg.go.dev/istio.io/istio/pkg/workloadapi#LoadBalancing_Scope
+1. Configure the load balance mode (failover or strict) and scope (REGION, ZONE, etc.) of the service. This corresponds to the message LoadBalancing in workload.proto.
+```
+message LoadBalancing {
+  enum Scope {
+    UNSPECIFIED_SCOPE = 0;
+    REGION = 1;
+    ZONE = 2;
+    SUBZONE = 3;
+    NODE = 4;
+    CLUSTER = 5;
+    NETWORK = 6;
+  }
+  enum Mode {
+    UNSPECIFIED_MODE = 0;
+    STRICT = 1;
+    FAILOVER = 2;
+  }
+  repeated Scope routing_preference = 1;
+  Mode mode = 2;
+}
+```
+
+Add scope and mode information to the service map, and incorporate corresponding update logic into the control plane.
+
+`bpf/kmesh/workload/include/workload.h`
+```
+typedef struct {
+    __u32 endpoint_count;               // endpoint count of current service
+    __u32 lb_policy;                    // load balancing algorithm
+    __u32 service_port[MAX_PORT_COUNT]; // service_port[i] and target_port[i] are a pair, i starts from 0 and max value
+                                        // is MAX_PORT_COUNT-1
+    __u32 target_port[MAX_PORT_COUNT];
+    struct ip_addr wp_addr;
+    __u32 waypoint_port;
+    __u8 balance_scope[MAX_SCOPE_COUNT]; // balance scope 
+    __u8 balance_mode;                   // balance mode: Unspecified_mode 0; strict 1; failover 2
+} service_value;
+```
+
+Add corresponding fields to the `pkg/controller/workload/bpfcache/service.go`, and update logic to `pkg/controller/workload/workload_processor.go`
+
+2. Configure the locality (region, zone, subzone) and health status （HEALTHY, UNHEALTHY）of the backend. This corresponds to the message in workload.proto.
+> Although the current workload API defines seven scopes, when configuring a pod's locality, only region, zone, and subzone are configured. Therefore, matching capabilities can only be realized for these three scopes.
```

Comment:
>only region, zone, and subzone are configured

where do you get this conclusion, at least NODE is supported now

Reply:
I misunderstood it before. I saw that in ztunnel, the NODE, NETWORK, and CLUSTER information were maintained within the workload, and I considering adding these informations to the bpf map of the backend later.